lib/chef/client.rb

```diff
@@ -957,6 +959,12 @@ def has_admin_privileges?
 
       Chef::ReservedNames::Win32::Security.has_admin_privileges?
     end
+
+    # Register the data collector reporter to send event information to the
+    # data collector server
+    def register_data_collector_reporter
+      events.register(Chef::DataCollector::Reporter.new) if Chef::DataCollector.register_reporter?
```

Comment:
`register_reporters` is not called if you're in Chef Solo legacy mode, and this feature needs to work with all run-types (Solo Legacy, Solo / Client Local Mode, and Client/Server mode). That's why we broke it up and call this method directly in `#run`.